src/Base/Matrix.cpp

```diff
@@ -832,37 +832,42 @@ ScaleType Matrix4D::hasScale(double tol) const
     // Only in case of uniform scaling it doesn't make a difference.
     if (tol == 0.0)
         tol = 1e-9;
+    
+    // compare absolute values propotional to size
+    auto close = [&](double a, double b) {
+        return fabs(fabs(a/b)-1.0) <= tol;
+    };

```

Comment:
Ah. Good point. When the scale was very large the tolerance didn't work well. Maybe I can scale the tolerance instead. Or just go back to absolute compare.

Reply:
Another weakness of this expression is that for a=1 and b=-1 it returns true.

For a relative comparison have a look at: https://c-faq.com/fp/fpequal.html

Reply:
@wwmayer That was intentional, to make it return Uniform even if scale is negative.

Reply:
OK, but then it might be good to choose a lambda name that makes it more obvious like closeAbsolute or so.

Reply:
Sure

Reply:
I wonder what to do with negative scaling because effectively this is a (positive) scaling and a mirroring. If the  analysis gives a number of negative scaling factors of 2 then it can be considered as a positive scaling with two mirrorings. But two mirroring operations can also be considered as a rotation.